### PR TITLE
[documentation] Add a section storage requirements

### PR DESCRIPTION
**Pull request by amontoison**
https://juliasmoothoptimizers.github.io/Krylov.jl/previews/PR666/storage/

### CODE REVIEW

**Comment by amontoison** [comment]
@dpo, It's for the last part of the JOSS paper.

File: docs/src/storage.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+|:-------:|:---------------:|:---------------:|:-------------------------------:|:-------------------------------------------:|
+ Storage  | $4n$            | $5n$            | $5n$                            | $3n + 2np + 5p$                             |
+
+#### Hermitian indefinite linear systems
+
+| Methods | [`SYMMLQ`](@ref symmlq) | [`MINRES`](@ref minres) | [`MINRES-QLP`](@ref minres_qlp) |
+|:-------:|:-----------------------:|:-----------------------:|:-------------------------------:|
+| Storage | $5n$                    | $6n$                    | $6n$                            |
+
+#### Non-Hermitian linear systems
```
Comment (dpo) [inline docs/src/storage.md:62]:
```suggestion
#### Non-Hermitian square linear systems
```

**Review by dpo — APPROVED** [review]
Looks great! Just a minor comment.

File: docs/src/storage.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+Thereafter, we can use `Base.format_bytes(nbytes)` to recover what is displayed in the REPL.
+
+```@example storage
+Base.format_bytes(nbytes)
+```
+
+To verify that we match the theoretical results, we just need to multiply the storage requirement of a method by the number of bytes associated to the precision of the linear problem.
+For instance, we need 4 bytes for the precision `Float32` , 8 bytes for precisions `Float64` and `ComplexF32`, and 16 bytes for the precision `ComplexF64`.
```
Comment (amontoison) [inline docs/src/storage.md:135]:
```suggestion
For instance, we need 4 bytes for the precision `Float32`, 8 bytes for precisions `Float64` and `ComplexF32`, and 16 bytes for the precision `ComplexF64`.
```

**Pull request merged**